helix-core/src/main/java/org/apache/helix/controller/rebalancer/util/RebalanceScheduler.java

```diff
@@ -138,7 +140,7 @@ public static void invokeRebalance(HelixDataAccessor accessor, String resource)
     PropertyKey key = accessor.keyBuilder().idealStates(resource);
     IdealState is = accessor.getProperty(key);
     if (is != null) {
-      if (!accessor.updateProperty(key, is)) {
+      if (!accessor.updateProperty(key, znRecord -> znRecord, is)) {
```

Comment:
I guess this might be hard for those who are not familiar with the accessor to understand. Please either comment with some detail, such as version based update, or just create a new call "idleUpdateProperty" or so.

Reply:
Prefer to leave a comment instead of having the complete class. Then that would be meaningless of using Java 8.

Reply:
+1 to have a comment, otherwise it could be confusing here.